app/page.tsx

```diff
@@ -87,6 +88,102 @@ let columnsShown = false;
+ * (ie. no duplicate message ids). The Looker message data is also cleaned up
+ * to match the message data objects from asrouter, remove any test messages,
+ * and update surface tags.
+ */
+async function mergeLookerData() {
+  const fs = require("fs");
+
+  // Existing message data
+  let result = fs.readFileSync(
+    "lib/asrouter-local-prod-messages/data.json",
+    "utf8",
+  );
+  let json_result = JSON.parse(result);
+
+  // Looker message data
+  let data = await runLookQuery();
+
+  // Manual check to remove messages with id defined inside `removeMessages`
+  let clean_data = data.filter((messageDef: any) => {
+    const removeMessages = [
+      "undefined",
+      "",
+      "test-id",
+      "n/a",
+      null,
+      "DEFAULT_ID",
+    ];
+    return !removeMessages.includes(
+      messageDef[
+        "messaging_system.metrics__text2__messaging_system_message_id"
+      ],
+    );
+  });
+  let json_looker_data = JSON.parse(JSON.stringify(clean_data));
+
+  // Add any message data with an id that does not already exist in data.json
+  for (let i = 0; i < json_looker_data.length; i++) {
+    // Check that id does not already exist and does not contain a "test" substring
+    if (
+      !json_result.find(
+        (x: any) =>
+          x.id ===
+          json_looker_data[i][
+            "messaging_system.metrics__text2__messaging_system_message_id"
+          ],
+      ) &&
+      !json_looker_data[i][
+        "messaging_system.metrics__text2__messaging_system_message_id"
+      ]
+        .toLowerCase()
+        .includes("test")
+    ) {
+      // Clean up objects to only include properties `id` and `template` since
+      // currently the other properties do not provide any value.
+      // NOTE: `hidePreview` is added because the message data from Looker
+      // does not have enough information to enable message previews correctly
+      // and we must manually hide the previews for now.
+      let clean_looker_object = {
+        id: json_looker_data[i][
+          "messaging_system.metrics__text2__messaging_system_message_id"
+        ],
+        template:
+          json_looker_data[i][
+            "messaging_system.metrics__string__messaging_system_ping_type"
+          ],
+        hidePreview: true,
+      };
+
+      // Update surfaces for RTAMO messages
+      if (clean_looker_object.id.includes("RTAMO")) {
+        clean_looker_object.template = "rtamo";
+      }
+
+      // Update surfaces for feature callout messages
+      if (clean_looker_object.template === null) {
+        clean_looker_object.template = "feature_callout";
+      }
+
+      // Update surface for FOCUS_PROMO message
+      if (clean_looker_object.id === "FOCUS_PROMO") {
+        clean_looker_object.template = "pb_newtab";
+      }
+
+      json_result.push(clean_looker_object);
+    }
+
+    fs.writeFileSync(
+      "lib/asrouter-local-prod-messages/data.json",
+      JSON.stringify(json_result),
+    );
```

Comment:
Once this gets hoisted into its own function, I think it's worth being more explicit about the various side-effects and the whys (i.e. this is effectively caching the looker data which will be generated at build time). Whether that's entirely what we want, I'm not yet sure, although I suspect it's ok. This is more of value to the development workflow than the production workflow.